models/src/main/scala/coop/rchain/models/EqualsM.scala

```diff
@@ -98,6 +99,12 @@ object EqualM extends EqualMDerivation {
   implicit val MatchEqual            = gen[Match]
 
   implicit val ConnectiveEqual = gen[Connective]
+  implicit def SignedEqual[A: EqualM] = new EqualM[Signed[A]] {
+    override def equal[F[_]: Sync](self: Signed[A], other: Signed[A]): F[Boolean] =
+      EqualM[A]
+        .ensuring((self.sigAlgorithm, self.sig) == (other.sigAlgorithm, other.sig))
```

Comment:
@marcin-rzeznicki is `ensuring` function OK to use here. For false cases it will throw assertion exception.
https://github.com/scala-native/scala-native/blob/master/scalalib/overrides-2.11/scala/Predef.scala#L169

Reply:
Why would you want to throw an exception here?

Reply:
@marcin-rzeznicki I don't want exception here. I was not sure if this exception is handled differently by the compiler.
Any idea what can be used instead of `EqualM[A].ensuring`?

Reply:
Fix in #2801.